tests/c/swt_module_clone.c

```diff
@@ -0,0 +1,49 @@
+// ignore-if: test "$YKB_TRACER" != "swt"
+// Run-time:
+//   env-var: YKD_LOG_IR=-:aot,jit-pre-opt
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YK_LOG=4
+//   status: success
+
+// Check that functions which address is taken can refer to other
+// functions which address is not taken. This test is specific for SWT
+// with module cloning enabled. Note that the cloned functions will not
+// be visible in the aot ir since they are not serialised.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int add(int i, int j) { return i + j; }
```

Comment:
These should probably be `__attribute__((yk_outline))`: at the moment I would expect that these are inlined, so the test will pass "by accident" (TM). Also, the `stderr` lang_test should check that the IR actually looks like we expect. Specifically, we should remove `jit-pre-opt` from `YKD_LOG_IR` but we should actually check that the functions, with the names we expect, are present in the AOT IR.

Reply:
OK, will update.
But there's a problem with adding IR checks here because we don't serialize the cloned/optimized functions at all.

Reply:
That's fine: if the AOT IR should only refer to the unoptimised functions, then that's our test!

Reply:
Updated :point_right: 6bf7c04fdd10dd811dfcda75e03acd1bcc1cbc1e and :point_right: 0dedb052da113a0e39ac097888b4de25acd14afb

Reply:
Should there be a `call dec` in the trace IR too?

Reply:
How do I get the trace IR?

Reply:
It depends which you want: `aot`, `jit-pre-opt`, or `jit-post-opt`.

Reply:
ok, got it.
I got confused cause originally we said to remove `jit-pre-opt` from YKD_LOG_IR

Reply:
> Should there be a call dec in the trace IR too?

Added trace checks :point_right: d4b95bfebbf30426952bc5db2e75be949748c92e

Reply:
Can we check the call is to `dec`? I'm not totally sure off-hand if that is possible or not.